homeassistant/components/velbus/__init__.py

```diff
@@ -138,9 +139,17 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     await hass.data[DOMAIN][entry.entry_id]["cntrl"].stop()
     hass.data[DOMAIN].pop(entry.entry_id)
+    await hass.async_add_executor_job(
+        shutil.rmtree,
+        hass.config.path(f".storage/velbuscache/{entry.entry_id}/"),
+    )
     if not hass.data[DOMAIN]:
         hass.data.pop(DOMAIN)
         hass.services.async_remove(DOMAIN, SERVICE_SCAN)
         hass.services.async_remove(DOMAIN, SERVICE_SYNC)
         hass.services.async_remove(DOMAIN, SERVICE_SET_MEMO_TEXT)
+        await hass.async_add_executor_job(
+            shutil.rmtree,
+            hass.config.path(".storage/velbuscache/"),
+        )
```

Comment:
What if there are multiple entries (for example, two entries), one is disabled, and the other is being removed... would the cache of the disabled entry be removed now too?

Reply:
would the hass.data[DOMAIN] be empty if the integration is disabled but not removed?

in this yes, then this would happen

Reply:
> would the hass.data[DOMAIN] be empty if the integration is disabled but not removed?

Sorry, I'm confused, this is not about removal at all 🤦 
This is about unloading.

Please note, the unloading isn't called when Home Assistant shuts down normally. Is it expected for the cache folder to survive restarts, but not integration reloading?

Reply:
yes the cache folder should survive a restart

the cache is implemented to make sure that hass restarts faster after the intital data loading, this initial data loading can take up to 5 minuts

Reply:
So why is it deleted on integration reload in that case?